packages/codec/lib/core.ts

```diff
@@ -33,10 +33,15 @@ const Web3Utils = require("web3-utils");
 export function* decodeVariable(
   definition: Ast.AstNode,
   pointer: Pointer.DataPointer,
-  info: Evm.EvmInfo
+  info: Evm.EvmInfo,
+  compilationId: string
```

Comment:
I don't think that's feasible, no.  Basically now everything that takes an AST node requires knowing what compilation it's from.  In this case it's needed for handling of user-defined types.  You do a `definitionToType` (which requires this to handle user-defined types), then you decode with that.  I don't really see a better alternative?  Honestly I think since it's in `codec` it's fine. :P

Reply:
Yeah okee!